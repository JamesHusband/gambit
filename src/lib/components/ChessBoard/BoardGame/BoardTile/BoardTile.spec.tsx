import { render, screen } from "@testing-library/react";
import { BoardTile } from ".";

jest.mock("../../../../utils", () => ({
  cn: (...args: string[]) => args.join(" "),
}));

describe("BoardTile", () => {
  it("should render", () => {
    const { container } = render(
      <BoardTile position="a1" rowIndex={0} colIndex={0} />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render a white tile", () => {
    render(<BoardTile position="a1" rowIndex={0} colIndex={0} />);
    expect(screen.getByTestId("board-tile")).toHaveClass("bg-[#eeeed2]");
  });
  it("should render a dark tile", () => {
    render(<BoardTile position="a2" rowIndex={0} colIndex={1} />);
    expect(screen.getByTestId("board-tile")).toHaveClass("bg-[#769656]");
  });
});
