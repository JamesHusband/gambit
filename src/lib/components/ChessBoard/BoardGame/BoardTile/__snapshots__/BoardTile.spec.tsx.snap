// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BoardTile should render 1`] = `
<div>
  <div
    class="flex items-center justify-center w-full h-full bg-[#eeeed2]"
    data-testid="board-tile"
  />
</div>
`;
